Userland/Libraries/LibJS/Runtime/Temporal/InstantPrototype.cpp

```diff
@@ -21,8 +23,42 @@ void InstantPrototype::initialize(GlobalObject& global_object)
 
     auto& vm = this->vm();
 
+    define_native_accessor(vm.names.epochSeconds, epoch_seconds_getter, {}, Attribute::Configurable);
+
     // 8.3.2 Temporal.Instant.prototype[ @@toStringTag ], https://tc39.es/proposal-temporal/#sec-temporal.instant.prototype-@@tostringtag
     define_direct_property(*vm.well_known_symbol_to_string_tag(), js_string(vm.heap(), "Temporal.Instant"), Attribute::Configurable);
 }
 
+static Instant* typed_this(GlobalObject& global_object)
+{
+    auto& vm = global_object.vm();
+    auto* this_object = vm.this_value(global_object).to_object(global_object);
+    if (!this_object)
+        return {};
+    if (!is<Instant>(this_object)) {
+        vm.throw_exception<TypeError>(global_object, ErrorType::NotA, "Temporal.Instant");
+        return {};
+    }
+    return static_cast<Instant*>(this_object);
+}
+
+// 8.3.3 get Temporal.Instant.prototype.epochSeconds, https://tc39.es/proposal-temporal/#sec-get-temporal.instant.prototype.epochseconds
+JS_DEFINE_NATIVE_FUNCTION(InstantPrototype::epoch_seconds_getter)
+{
+    // 1. Let instant be the this value.
+    // 2. Perform ? RequireInternalSlot(instant, [[InitializedTemporalInstant]]).
+    auto* instant = typed_this(global_object);
+    if (vm.exception())
+        return {};
+
+    // 3. Let ns be instant.[[Nanoseconds]].
+    auto& ns = instant->nanoseconds();
+
+    // 4. Let s be RoundTowardsZero(ℝ(ns) / 10^9).
+    auto [s, _] = ns.big_integer().divided_by(Crypto::SignedBigInteger::create_from(1'000'000'000));
+
+    // 5. Return 𝔽(s).
+    return Value((double)s.to_base(10).to_int<i64>().value());
```

Comment:
how about:
```suggestion
    return Value((double)s.to_u64());
```

Reply:
u64 != i64 (I actually tried that and spent 10 minutes figuring out why the values are garbage :D)

Reply:
hmm, this is a positive value though, so that should be equivalent to `s.unsigned_value().to_u64()` no? (and while youre there, renaming it from the confusing name i chose might be nice :)

Reply:
the instant nanoseconds value is a signed bigint, so the division result can also be negative.

Reply:
oh you can make negative instants? thats a bit weird, never mind then, we can optimize this in the future

Reply:
I can even tell you how negative :D

```js
> new Temporal.Instant(-8_640_000_000_000_000_000_000n).toString()
"-271821-04-20T00:00:00Z"
```

(from the polyfill, no toString here yet)